### ISSUE TITLE
Azure portal for Graph Cosmos database does not visualize DataContract properties?

### ISSUE THREAD

**Issue by steevcoco** [Azure/azure-documentdb-dotnet#437]
Hi folks. I am new; but it seems that if I push an object into Cosmos, which has been set up with the Graph API --- and, this object is created as a `DataContract` object, and pushed with the .Net `DocumentClient` --- then in the Azure portal, the Graph node appears with the `id`, and `label` is always `"NativeVertex"`. And none of the object properties appear ... Is it some limitation, or something that I am doing that prevents me from being able to see object properties there?

The graph is always like this:

![image](https://user-images.githubusercontent.com/22354363/36326776-51061184-132a-11e8-8e55-cd1f2a064ab4.png)

... The properties DO deserialize and come back out. I just can't see them in the portal.

This object is like:

    [DataContract]
    public sealed class AdProduct
            : SubjectBase
    {
        [DataMember]
        public string SquareSku { get; private set; }
    }
    
    [DataContract]
    public class SubjectBase
            : HashSignatureBase,
                    ISubjectBase
    {
        [DataMember]
        public string Name { get; private set; }

        [DataMember]
        public string Uri { get; private set; }
    }
    
    [DataContract]
    public abstract class HashSignatureBase
            : HashSignature,
                    IHasId
    {
        [DataMember(Name ="id")]
        public string Id { get; private set; }
    }

**Comment by olivertowers** [comment]
**Short answer:** Graph explorer (and Gremlin) can only recognize vertices/edges and their properties if the backing Document matches the expected schema.

**Long answer:**
The primary usage pattern for CosmosDB Graph is to issue Gremlin queries via the Gremlin server endpoint. The Gremlin server reads/writes to the CosmosDB document storage backend and adheres to the schema requirements for representing graph elements as documents. Assuming you have written vertices and edges via Gremlin, then Graph explorer will recognize the elements in full.

The gremlin endpoint for your account can be found on overview page of your CosmosDB account on Azure portal, and a sample of how to connect to the server is [here](https://github.com/Azure-Samples/azure-cosmos-db-graph-gremlindotnet-getting-started).

Note that Gremlin server responses use GraphSON format to represent Vertices/Edges, see reference [here](http://tinkerpop.apache.org/docs/3.2.4/dev/io/#_graph_structure), which does not map directly to the POCO examples you have given.

Can you provide some more details on your use case for Graph and how you would use it in conjunction with the classes?
Based on this I can give suggestions on the best approaches.

**Comment by steevcoco** [comment]
OK ... Well that was an abundantly helpful reply @olivertowers !

I am new. I will be creating and storing objects like the above: all `DataContract`. I THOUGHT I would define the Cosmos DB as Graph because of the object relationships. --- Not shown above, I will have 'nested' objects within objects, just trying to work as much in C# as possible: some object will have a public `DataMember` property that is some other `DataContract` object ... And my usage will seem to be primarily through the C# `DocumentClient`.

I really just want to work with objects in C# and put and pull them from the database like seamless magic! Some objects will hold `IReadOnlyCollection` lists of other (`DataContract`) objects ... But it is not a complex data model; and relations are just simple containment.

It is primarily just a "bag" of objects. I need to fetch them by `ID`, or by some simple property value; and the collection will not be huge (humiliating): I will just need to find Users by Email or Id, add and remove Subscriptions and update the User in-place; and add and remove Licenses. If the collection grows past 4,000 objects, a dream is coming true, and I can re-work something!

    [DataContract]
    public class User
    {
        [DataMember(Name ="id")]
        public string Id { get; private set; }
		
        [DataMember]
        public string Name { get; private set; }
		
        [DataMember]
        public string Email { get; private set; }
		
        [DataMember]
        public string ClientId { get; private set; }
		
        [DataMember]
        public IReadOnlyCollection<Subscription> Subscriptions { get; private set; }
    }
    
    [DataContract]
    public class Subscription
    {
        [DataMember(Name ="id")]
        public string Id { get; private set; }
		
        [DataMember]
        public string UserId { get; private set; }
		
        [DataMember]
        public string Name { get; private set; }
		
        [DataMember]
        public string ProductName { get; private set; }
		
        [DataMember]
        public string ProductUri { get; private set; }
		
        [DataMember]
        public string ProductId { get; private set; }
		
        [DataMember]
        public TimeSpan Duration { get; private set; }
		
        [DataMember]
        public string TransactionId { get; private set; }
		
        [DataMember]
        public License License { get; private set; }
		
        [DataMember]
        public DateTime? ActivatedAt { get; private set; }
    }
    
    [DataContract]
    public class License
    {
        [DataMember(Name ="id")]
        public string Id { get; private set; }
		
        [DataMember]
        public string UserId { get; private set; }
		
        [DataMember]
        public string ProductId { get; private set; }
		
        [DataMember]
        public DateTime ExpirationTime { get; private set; }
    }

**Comment by steevcoco** [comment]
@olivertowers My humiliation not withstanding, I just wanted to note that I have made some changes and moved forward ... I have actually switched the Cosmos database to the Sql API, and things are working more smoothly. I wasn't actually benefiting from the Graph API, and I suppose I thought I might benefit from some more "plain object containment" model, but I'm not using the actual Graph API so it went nowhere. The SQL API is very nice.
